### DISCUSSION TITLE
Recommended compiler options

### DISCUSSION THREAD

**Discussion by goshoo** [awslabs/aws-sdk-rust#644] [General]
Hi,

I compiled slightly modified version of this Lambda example - https://docs.aws.amazon.com/sdk-for-rust/latest/dg/lambda.html

with:

$  cross build --release --target x86_64-unknown-linux-musl

and the output file size was ~18MB

When I added the following options to Cargo.toml

```
[profile.release]
strip = "symbols"
opt-level = 'z'
lto = true
codegen-units = 1
```

file size changed to ~4.5MB

So my questions are - are these options safe to be used or do you recommend better/safer options?

Thanks in advance.

**Comment by jdisanti** [comment]
The only thing I would be wary of is `strip = "symbols"`. While that will get you a smaller file size, I think it will also remove function names in the event of a backtrace being emitted for a panic. The rest should be fine.

**Reply by goshoo** [reply]
Thank you, @jdisanti , for recommendation!
